### ISSUE TITLE
Unable to sendmail after upgrade from 1.3.9 to 1.4.2

### ISSUE THREAD

**Issue by jasonmacer** [roundcube/roundcubemail#7150]
First off, love the new layout!  

I performed an upgrade as I was unable to upload attachments in 1.3.9. Granted, I can upload them now, however when I go to send the email I get the following error when sending (with AND without an attachment):  
SMTP Error (-1): Connection to the server failed.

tail -f ../roundcubemail/logs/errors shows the following:

> [07-Jan-2020 14:27:53 -0600]: DB Error: Configuration error. Unsupported database driver: mysql in ../roundcubemail/program/lib/Roundcube/rcube_db.php on line 83 (POST /webmail/?_task=help&_action=refresh)

So naturally the first place i look is in ../roundcubemail/config/config.inc.php at my connection string. I have edited it for security purposes:

>$config['db_dsnw'] = 'mysql://roundcubeuser:************@localhost/roundcubedb';

This looks correct to me, so then I went over to ../roundcubemail/program/lib/Roundcube/rcube_db.php to have a look, and to the best of my ability line 83 falls in the middle of the public static function:

```
public static function factory($db_dsnw, $db_dsnr = '', $pconn = false)
    {
     	$driver     = strtolower(substr($db_dsnw, 0, strpos($db_dsnw, ':')));
        $driver_map = array(
            'sqlite2' => 'sqlite',
            'sybase'  => 'mssql',
            'dblib'   => 'mssql',
            'mysqli'  => 'mysql',
            'oci'     => 'oracle',
            'oci8'    => 'oracle',
        );

	$driver = isset($driver_map[$driver]) ? $driver_map[$driver] : $driver;
        $class  = "rcube_db_$driver";

        if (!$driver || !class_exists($class)) {
            rcube::raise_error(array('code' => 600, 'type' => 'db',  ### THIS SEEMS TO BE LINE 83###
                'line' => __LINE__, 'file' => __FILE__,
                'message' => "Configuration error. Unsupported database driver: $driver"),
                true, true);
        }

	return new $class($db_dsnw, $db_dsnr, $pconn);
    }
```

The only difference I can see is that my connection string begins with 

>$config['db_dsnw'} = 'mysql:// 

This compared to the array above which has 

>'mysqli' => 'mysql',

I did try changing my 'mysql://... to 'mysqli://... and still I got the error.

Sorry to be so obtuse here, but what am i missing? Do I need to restart the server and ruin my 6 month uptime?

Thanks for any assistance!

Roundcube Version:
>Roundcube Webmail 1.4.2

Here is the list of active plugins:
>$config['plugins'] = array('additional_message_headers', 'archive', 'attachment_reminder', 'autologon', 'database_attachments', 'debug_logger', 'emoticons', 'enigma', 'filesystem_attachments', 'help', 'hide_blockquote', 'http_authentication', 'identicon', 'identity_select', 'jqueryui', 'managesieve', 'markasjunk', 'new_user_dialog', 'new_user_identity', 'newmail_notifier', 'password', 'show_additional_headers', 'subscriptions_option', 'userinfo', 'vcard_attachments', 'zipdownload');

**Comment by alecpl** [comment]
Probably broken upgrade. Do you have `program/lib/Roundcube/db/mysql.php` file?

ps. remove one of database_attachments and filesystem_attachments from the plugins list. It will not gonna work.

**Comment by jasonmacer** [comment]
@alecpl indeed I do, oh and you're the author. Nice!

I should also mention that the upgrade script did say that it completed successfully.

**Comment by alecpl** [comment]
Maybe it can't access the config files or whole folder. I don't know, it's strange.

**Comment by jasonmacer** [comment]
That's an interesting thought. 

Just an ls -l of the files inside the config/ directory show they are all owned by root:root

backing up to the main roundcubemail directory shows that 

public_html/ && config/ = root:root
logs/ && temp/ = apache:apache
and everything else is 501:80

the roundcubemail/ directory is root:root

I don't know if this is how it is *supposed* to be, or if some of these settings changed with the update. I did make a copy of the roundcubemail folder before the upgrade and it is root:root, as is everything inside there is root:root as well.

dare i try chown apache:apache on the whole folder and all subfolders and files?

**Comment by alecpl** [comment]
It can be root:root if you allow apache to read all these files. logs and temp need to be writeable.

**Comment by jasonmacer** [comment]
Yeah, logs and temp are writable.

**Comment by jasonmacer** [comment]
@alecpl, I'll look at performing a re-install this afternoon. I shouldn't have to update the DB but I'll create a new one just in case.

**Comment by jasonmacer** [comment]
@alecpl Found the issue, and it took a full re-install and some looking to figure it out. 

I performed a clean install however it still wouldn't send an email. 

It's been quite some time since I setup this server so I did a little netstat digging and realized my postfix server is running on port 25 for smtp not the default 587 that the installer is looking for now. 

Once I added the variable in config.inc.php it sent mail like a champ. That said, i was able to also restore the "upgraded" version and it worked too.

Live and learn \o/
